docs/sphinx/user_guide/feature/reduction.rst

```diff
@@ -159,3 +159,151 @@ Reduction Policies
 For more information about available RAJA reduction policies and guidance
 on which to use with RAJA execution policies, please see 
 :ref:`reducepolicy-label`.
+
+--------------------------------
+Experimental Reduction Interface
+--------------------------------
+
+An experimental reduction interface is now available that offers several 
+usability and performance advantages over the current reduction model in RAJA. 
+The new interface allows ``RAJA::forall`` to take optional "plugin-like" 
+objects to extend the execution behavior of a ``RAJA::forall`` execution 
+context.
+
+The new interface passes ``RAJA::expt::Reduce<OP_TYPE>`` objects as function 
+arguments to ``RAJA::forall`` and provides users with thread-local variables
+of the reduction data type to be updated inside the lambda. This differs 
+from the current reduction model in which ``RAJA::ReduceOP<REDUCE_POL, T>``
+objects are captured by the user-supplied kernel body lambda expression.
+
+
+RAJA::expt::Reduce
+..................
+::
+
+  double* a = ...;
+
+  double rs = 0.0;
+  double rm = 1e100;
+      
+  RAJA::forall<EXEC_POL> ( Res, Seg, 
+  RAJA::expt::Reduce<RAJA::operators::plus>(&rs),
+  RAJA::expt::Reduce<RAJA::operators::minimum>(&rm),
+  [=] (int i, double& _rs, double& _rm) {
+    _rs += a[i];
+    _rm = RAJA_MIN(a[i], _rm); 
+  }
+  );
+  
+  std::cout << rs ...
+  std::cout << rm ...
+
+* Each ``RAJA::expt::Reduce`` argument to ``RAJA::forall`` is templated on
+  a reduction operator, and takes a reference to a target variable to write 
+  the final reduction result to, ``&rs`` and ``&rm`` in the example code
+  above. The reduction operation will include the initial value of
+  the given target variable.
+* The kernel body lambda expression passed to ``RAJA::forall`` requires an 
+  argument corresponding to each ``RAJA::expt::Reduce`` argument, ``_rs`` and 
+  ``_rm`` in the example code. These arguments are used as a local instance of 
+  of each target of a reduction operation. It is important to note that the 
+  the lambda expression reduction arguments must follow the kernel iteration
+  variable, ``i`` in this case, and appear in the same order as the 
+  corresponding ``RAJA::expt::Reduce`` arguments to ``RAJA::forall``. They
+  also must have the same types as the values passed to the 
+  ``RAJA::expt::Reduce`` arguments. 
+* The local variables ``_rs`` and ``_rm`` are initialized with the *identity* of
+  the reduction operation to be performed.
+* A reduction is performed implicitly by the ``RAJA::forall`` execution
+  across thread copies of the local variables.
+* Finally, the reduction operation is performed against the original value of 
+  the target and the result of the ``RAJA::forall`` reduction.
+* The final reduction value is accessed by referencing the target variable 
+  passed to ``RAJA::expt::Reduce`` in the ``RAJA::forall`` method.
```

Comment:
```suggestion
* The target variable passed to the ``RAJA::expt::Reduce`` argument
  of ``RAJA::forall`` then has the final reduction value.
```

Reply:
I'm not sure about this one, but I took the rest